lib/handle_publish.c

```diff
@@ -135,7 +135,9 @@ int handle__publish(struct mosquitto *mosq)
 			return MOSQ_ERR_SUCCESS;
 		case 1:
 			util__decrement_receive_quota(mosq);
-			rc = send__puback(mosq, mid, 0, NULL);
+			if(mosq->delayed_puback){
```

Comment:
should the ifs not the other way around?

delayed_puback is per default false so it should still to the send__puback here by default, no?

Reply:
yes 😁